android/quest/src/main/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireActivity.kt

```diff
@@ -336,26 +339,75 @@ class QuestionnaireActivity : BaseMultiLanguageActivity() {
               questionnaireConfig = questionnaireConfig,
               actionParameters = actionParameters,
               context = this@QuestionnaireActivity,
-            ) { idTypes, questionnaireResponse ->
+            ) { idTypes, questionnaireResponse, extractedValidationErrors ->
               // Dismiss progress indicator dialog, submit result then finish activity
               // TODO Ensure this dialog is dismissed even when an exception is encountered
               setProgressState(QuestionnaireProgressState.ExtractionInProgress(false))
-              setResult(
-                Activity.RESULT_OK,
-                Intent().apply {
-                  putExtra(QUESTIONNAIRE_RESPONSE, questionnaireResponse as Serializable)
-                  putExtra(QUESTIONNAIRE_SUBMISSION_EXTRACTED_RESOURCE_IDS, idTypes as Serializable)
-                  putExtra(QUESTIONNAIRE_CONFIG, questionnaireConfig as Parcelable)
-                },
-              )
-              finish()
+
+              if (BuildConfig.BUILD_TYPE.contains("debug", ignoreCase = true)) {
+                val message =
+                  if (extractedValidationErrors.isEmpty()) {
+                    "Questionnaire submitted and saved successfully with no validation errors"
+                  } else {
+                    """
+                    Questionnaire `${questionnaire?.referenceValue()}` was submitted but had the following validation errors
+                    
+                    ${
+                                            buildString {
+                                                extractedValidationErrors.forEach {
+                                                    appendLine("${it.key}: ")
+                                                    append(it.value)
+                                                    appendLine()
+                                                }
+                                            }
+                                        }
+                                    """
+                      .trimIndent()
+                  }
+
+                AlertDialogue.showInfoAlert(
```

Comment:
From the UX perspective, by the time this dialog is shown, the QuestionnaireResponse has been used anyway. By the time the activity is finished with the result, all the other processes based on the QR have been completed. How do we intend to use the validation error messages? Will the messages only be relevant to devs/QA during testing? @dubdabasoduba do you have any comments on this?